v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/edr-provider.ts

```diff
@@ -283,66 +283,61 @@ export class EdrProvider extends BaseProvider {
       }
 
       if (stackTrace !== null) {
-        error = encodeSolidityStackTrace(
-          jsonRpcResponse.error.message,
+        // Pass data and transaction hash from the original error
+        /* eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        -- If we have a stack trace, we know that the json rpc response data
+        is an object with the data and transactionHash fields */
+        const responseErrorData = responseError.data as {
```

Comment:
Let's assert this, as this field has a lot of historic nuance, and if an assumption is incorrect, we should detect it and fix it asap.

Reply:
Also, note for @fvictorio @nebasuke: while this error data is now returned as `JsonRpcReponse.data`, it could as well be part of our own metadata object, which would allow us to have stronger type guarantees.